### ISSUE TITLE
Thread is missing in thread list 

### ISSUE THREAD

**Issue by gharlan** [mnapoli/externals#123]
https://externals.io/message/107388
This thread is missing in thread list (https://externals.io)

**Comment by mnapoli** [comment]
That's right, I don't have an explanation at the moment (I haven't looked much but the root cause doesn't look obvious right now).

**Comment by gharlan** [comment]
I guess it looks like this:

1. The mail has In-Reply-To and References header with non existing id (`5d976915.1c69fb81.d89d5.5e2fSMTPIN_ADDED_MISSING@mx.google.com`)
2. the threadId is read from References [without validation](https://github.com/mnapoli/externals/blob/e22aa15db60840665bc67e0f9311d0e198ae40bc/src/EmailSynchronizer.php#L165-L194)
3. while inserting the mail, [`isThreadRoot` is set](https://github.com/mnapoli/externals/blob/e22aa15db60840665bc67e0f9311d0e198ae40bc/src/Email/EmailRepository.php#L188) to `false`, because [threadId != mail id](https://github.com/mnapoli/externals/blob/e22aa15db60840665bc67e0f9311d0e198ae40bc/src/Email/Email.php#L140)

**Comment by francislavoie** [comment]
At this point, there's a few entire threads missing, I've had to revert to reading https://news-web.php.net/php.internals 😢 

I would guess that it's related to this change https://github.com/mnapoli/externals/pull/120 which may not be robust enough to handle certain mails as @gharlan said.

**Comment by mnapoli** [comment]
Thanks for investigating @gharlan!

🤔 that's really annoying indeed, I'm not really sure what to do here (I unfortunately have a lot to deal with at the moment so it's hard to sit down and investigate properly). Do you see a solution?
